File: 367_valid_perfect_square.java
class Solution {
    public boolean isPerfectSquare(int num) {
         int ld=num%10;
       
         if(ld==3||ld==7||ld==2||ld==8){
            return false;
        }
      int l=1;
      int r=num;
        while(l<=r){
            long  mid=l+(r-l)/2;
            long sq=mid*mid;
            if(sq==num){
                return true;
            }
            else if(sq<num){
                l=(int)mid+1;
            }else{
                r=(int)mid-1;
            }
        }
        return false;
    }
}

File: 367_valid_perfect_square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        ld=num%10
        
        if(ld==3 or ld==7 or ld==2 or ld==8):
            return False
        l=1
        r=num
        while(l<=r):
            mid=int(l+(r-l)/2)
            sq=mid*mid;
            if(sq==num):
                return True
            elif(sq<num):
                l=int(mid+1)
            else:
                r=int(mid-1)
        return False
        
